src/codegen/doc/format.rs

```diff
@@ -0,0 +1,74 @@
+use regex::{Captures, Regex};
+
+const LANGUAGE_SEP_END : &'static str = "\" -->";
+const LANGUAGE_BLOCK_BEGIN : &'static str = "|[<!-- language=\"";
+const LANGUAGE_BLOCK_END : &'static str = "\n]|";
+
+lazy_static! {
+    static ref REG : Regex = Regex::new(r"#?(?:G[A-Z][a-z]|G[dt]k)[\w]*").unwrap();
+    static ref REG2 : Regex = Regex::new(r"@\w*(\s|\n)").unwrap();
+}
+
+pub fn reformat_doc(input: &str) -> String {
+    let out = code_blocks_transformation(input);
+    replace_c_types(&out)
+}
+
+fn code_blocks_transformation(input: &str) -> String {
+    let mut out = String::new();
+
+    for (num, entry) in input.split(LANGUAGE_BLOCK_BEGIN).into_iter().enumerate() {
+        if num & 1 == 0 {
+            out.push_str(&replace_c_types(entry));
+            continue;
+        }
+        out.push_str("```");
+        for (num, entry) in get_language(entry, &mut out).split(LANGUAGE_BLOCK_END)
+                                                         .into_iter()
+                                                         .enumerate() {
+            if num & 1 == 0 {
+                out.push_str(entry);
+                continue;
+            }
+            out.push_str(&format!("\n```{}", replace_c_types(entry)));
+            break;
+        }
+    }
+    out
+}
+
+fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
+    if let Some(pos) = entry.find(LANGUAGE_SEP_END) {
+        out.push_str(&entry[0..pos]);
+        &entry[(pos + LANGUAGE_SEP_END.len())..]
+    } else {
+        out.push_str("\n");
+        entry
+    }
+}
+
+fn replace_c_types(entry: &str) -> String {
+    let out = &REG.replace_all(entry, |caps: &Captures| {
+        match caps.at(0) {
+            Some(mut s) => {
+                if s.starts_with("#") {
+                    s = &s[1..];
+                }
```

Comment:
And this is why `G[A-Z][a-z]` is not the best approach. It might be possible to get more creative in the "substance" part, e.g. `(G|G[dt]k)([A-Z]\w*[a-z]\w*)` so the prefix must be followed by an uppercase letter but there also must be at least one lowercase one there somewhere.